src/journal-remote/journal-upload-journal.c

```diff
@@ -251,17 +251,44 @@ static void check_update_watchdog(Uploader *u) {
 
 static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void *userp) {
         Uploader *u = ASSERT_PTR(userp);
+        _cleanup_free_ char *compression_buffer = NULL;
         int r;
         sd_journal *j;
         size_t filled = 0;
         ssize_t w;
 
         assert(nmemb <= SSIZE_MAX / size);
 
+        if (u->tmp_size) {
```

Comment:
Please add a short comment what this code block does. It is hard to understand why this is inserted.

Reply:
it's related to https://github.com/systemd/systemd/pull/34822#discussion_r1929595285
will remove it if it's not needed, but would like to know you collective opinion as sometimes your and @poettering opinions are contradicting and i have no idea what should i do